pkg/planner/cascades/util/string_writer.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2024 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"bufio"
+	"io"
+
+	"github.com/pingcap/tidb/pkg/util/intest"
+)
+
+// StrBufferWriter is interface facilitate quick writing string regardless of error handling and written len.
+type StrBufferWriter interface {
+	WriteString(s string)
+	Flush()
+}
+
+// StrBuffer is a basic wrapper bufio.Writer while override its WriteString func.
+type StrBuffer struct {
+	bio *bufio.Writer
+}
+
+// NewStrBuffer new a defined buffed string writer with passed io.writer.
+func NewStrBuffer(w io.Writer) StrBufferWriter {
+	return &StrBuffer{
+		bio: bufio.NewWriter(w),
+	}
+}
+
+// WriteString implements IBufStrWriter
+func (sw *StrBuffer) WriteString(s string) {
```

Comment:
Why do we need an extra package for strBufferWriter ? Just for add a assert in the test ?

Reply:
yes, it's for simplicity. otherwise, we should add hint skipper like the below

Reply:
![image](https://github.com/user-attachments/assets/a11de884-a274-49a0-a304-f0d8dde29b7d)